libfreerdp/core/gateway/rdg.c

```diff
@@ -298,21 +323,128 @@ static BOOL rdg_write_packet(rdpRdg* rdg, wStream* sPacket)
 	return TRUE;
 }
 
-static BOOL rdg_read_all(rdpTls* tls, BYTE* buffer, int size)
+static int rdg_socket_read(BIO* bio, BYTE* pBuffer, size_t size,
+                           rdg_http_encoding_context* encodingContext)
+{
+	if (encodingContext->httpTransferEncoding == TransferEncodingIdentity)
+		return BIO_read(bio, pBuffer, size);
+
+	if (encodingContext->httpTransferEncoding == TransferEncodingChunked)
+	{
+		assert(encodingContext != NULL);
```

Comment:
Please no function calls before variable declaration, old compilers do not like this. (hello vs2010)